File: duolingo_rum/Models/UserAchievement.cs
using System;
using System.Collections.Generic;

namespace duolingo_rum.Models;

public partial class UserAchievement
{
    public int Id { get; set; }

    public Guid UserId { get; set; }

    public int AchievementId { get; set; }

    public DateTime? EarnedAt { get; set; }

    public virtual Achievement Achievement { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: android-app/app/src/main/java/com/linguaai/app/models/UserAchievement.kt
package com.linguaai.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UserAchievement(
    val id: Int = 0,
    @SerialName("user_id")
    val userId: String = "",
    @SerialName("achievement_id")
    val achievementId: Int = 0,
    @SerialName("earned_at")
    val earnedAt: String? = null
)
